include/Link.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "NetworkDevice.hpp"
+
+#include <cstdint>
+#include <memory>
+
+class Link {
+public:
+    Link(std::shared_ptr<NetworkDevice> source, std::shared_ptr<NetworkDevice> destination, std::uint64_t distance_ns);
+
+    void UpdateLastProcessTime(std::uint64_t new_process_time_ns);
+
+    std::shared_ptr<NetworkDevice> source;
+    std::shared_ptr<NetworkDevice> destination;
+    std::uint64_t distance_ns;
+
+    // In bits per ns
+    std::uint64_t speed = 0;
```

Comment:
I'd suggest to use `bandwidth` word instead of `speed`. Here's an explanation: `bandwidth` specifically refers to the maximum rate of data transfer measured in bits per second, accurately reflects the capacity of a network link, distinguishes itself from latency - which refers to the time delay in data transmission - emphasizes the relevance of data capacity for application performance, and **is widely used in networking literature and discussions**.